### ISSUE TITLE
Julia v1.0 Optim fails on precompile

### ISSUE THREAD

**Issue by braamvandyk** [JuliaNLSolvers/Optim.jl#673]
Clean install of Julia v1.0
```
Julia Version 1.0.0
Commit 5d4eaca0c9 (2018-08-08 20:58 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-7600U CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)
Environment:
  JULIA_HOME = C:\MyApps\Julia-1.0.0\bin
```
Installed packages:

```
    Status `C:\Users\vdykb\.julia\environments\v1.0\Project.toml`
  [c52e3926] Atom v0.7.3
  [6e4b80f9] BenchmarkTools v0.4.0
  [a93c6f00] DataFrames v0.13.0
  [39dd38d3] Dierckx v0.3.0
  [0c46a032] DifferentialEquations v5.2.0
  [31c24e10] Distributions v0.16.2
  [8f5d6c58] EzXML v0.8.0
  [7073ff75] IJulia v1.9.2
  [a98d9a8b] Interpolations v0.8.0
  [42fd0dbc] IterativeSolvers v0.7.0
  [033835bb] JLD2 v0.1.0
  [e5e0dc1b] Juno v0.5.2
  [76087f3c] NLopt v0.5.0
  [429524aa] Optim v0.15.3
  [438e738f] PyCall v1.17.1
  [d330b81b] PyPlot v2.6.0
  [1fd47b50] QuadGK v2.0.0
  [fcd29c91] QuantEcon v0.14.2
  [f2b01f46] Roots v0.7.0
```
Minimal (non-)working example:

```julia
julia> using Optim
[ Info: Precompiling Optim [429524aa-4258-5aef-a3af-852621145aeb]
ERROR: LoadError: LoadError: LoadError: UndefVarError: @printf not defined
Stacktrace:
 [1] top-level scope
 [2] include at .\boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at .\loading.jl:1038
 [4] include at .\sysimg.jl:29 [inlined]
 [5] include(::String) at C:\Users\vdykb\.julia\packages\Optim\5JqO8\src\Optim.jl:18
 [6] top-level scope at none:0
 [7] include at .\boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at .\loading.jl:1038
 [9] include(::Module, ::String) at .\sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at .\boot.jl:319 [inlined]
 [12] eval(::Expr) at .\client.jl:389
 [13] top-level scope at .\none:3
in expression starting at C:\Users\vdykb\.julia\packages\Optim\5JqO8\src\types.jl:66
in expression starting at C:\Users\vdykb\.julia\packages\Optim\5JqO8\src\types.jl:64
in expression starting at C:\Users\vdykb\.julia\packages\Optim\5JqO8\src\Optim.jl:106
ERROR: Failed to precompile Optim [429524aa-4258-5aef-a3af-852621145aeb] to C:\Users\vdykb\.julia\compiled\v1.0\Optim\R5uoh.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] macro expansion at .\logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at .\loading.jl:1184
 [4] _require(::Base.PkgId) at .\logging.jl:311
 [5] require(::Base.PkgId) at .\loading.jl:852
 [6] macro expansion at .\logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at .\loading.jl:834
```

**Comment by anriseth** [comment]
Optim version 0.15 does not support Julia 1.0, however, release 0.16 has
been released and supports Julia 1.0.

You may have ended up with an old version due to an upper bound in one of
your other packages?

(I think someone else had this issue somewhere, but don't have sufficient
reception to search for the violating package)

**Comment by pkofod** [comment]
I think some packages are being upper bounded to avoid stuff like this. Please ask on discourse to help find the offending package.
